### PR TITLE
Update storage implementation +  usecases

### PR DESCRIPTION
**Pull request by sheehui**
explanation on how asset/liabilities/policies are stored + use cases

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -415,6 +434,116 @@ State: Client Window (Contact Tab)
     * 1b1. Contact is not updated and DonnaFin shows an error message. \
       Use case resumes at step 1.
 
+**UC10: Adding an Asset to a Client**
```
Comment (bharathcs) [inline docs/DeveloperGuide.md:437]: I noticed you some times use Title case for 'Client' and sometimes lower case 'clients' for UC. same with other words here like 'asset', 'policy,  and  'home window'. I would not use title case for these as we're talking about nouns (and not proper nouns or class names), but up to you, as long as its consistent!

Comment (sheehui) [reply docs/DeveloperGuide.md]: @4d62dc3

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -198,6 +198,18 @@ The `Storage` component,
...
+* Each `Person` in DonnaFin is stored as a `JsonAdaptedPerson` as shown in our class diagram above.
+* The name, email, phone number, address and notes of each `Person` is stored as a JsonProperty of type String.
+* Assets, liabilities and policies are stored as a List of `JsonAdaptedAsset`, `JsonAdaptedLiability`
+and `JsonAdaptedPolicy` respectively. Each `JsonAdaptedAsset`, `JsonAdaptedLiability`
+and `JsonAdaptedPolicy` contains the necessary fields for their respective attributes, stored as a JsonProperty of type String.
+* As a `Person` can have more than one `Asset`, `Liability` or `Policy`, we have chosen to store them as nested objects rather
+converting them into a single String before storage and converting them back to the correct attributes on retrieval from the database.
+* As such, `Asset`, `Liability` and `Policy` are stored in this special manner and any future attributes that share the same characteristics
+can be stored in the same way.
+
```
[the remark this answers is not in the document]
Comment (sheehui) [reply docs/DeveloperGuide.md:213]: @4d62dc3

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -198,6 +198,18 @@ The `Storage` component,
 * inherits from both `AddressBookStorage` and `UserPrefStorage`, which means it can be treated as either one (if only the functionality of only one is needed).
 * depends on some classes in the `Model` component (because the `Storage` component's job is to save/retrieve objects that belong to the `Model`)
 
+How is `Person` stored?
+* `JsonAdaptedPerson` is created with Jackson.
+* Each `Person` in DonnaFin is stored as a `JsonAdaptedPerson` as shown in our class diagram above.
+* The name, email, phone number, address and notes of each `Person` is stored as a JsonProperty of type String.
+* Assets, liabilities and policies are stored as a List of `JsonAdaptedAsset`, `JsonAdaptedLiability`
+and `JsonAdaptedPolicy` respectively. Each `JsonAdaptedAsset`, `JsonAdaptedLiability`
+and `JsonAdaptedPolicy` contains the necessary fields for their respective attributes, stored as a JsonProperty of type String.
```
[the remark this answers is not in the document]
Comment (sheehui) [reply docs/DeveloperGuide.md]: @4d62dc3

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -201,17 +201,48 @@ The `Storage` component,
...
+    "name" : "HDB @Jurong",
+    "type" : "Property",
+    "value" : "$ 300000.00",
+    "remarks" : "BTO"
+  } ]
+}
+```
 
+### 4.6 Common classes
+2
```
Comment (bharathcs) [inline docs/DeveloperGuide.md:245]:
```suggestion

```

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -198,8 +198,50 @@ The `Storage` component,
...
+  "assets" : [ {
+    "name" : "HDB @Jurong",
+    "type" : "Property",
+    "value" : "$ 300000.00",
+    "remarks" : "BTO"
+  } ]
+}
+```
 
+### 4.6 Common classes
```
Comment (bharathcs) [inline docs/DeveloperGuide.md:244]:
```suggestion
### 4.6 Common classes

```

**Review by bharathcs — APPROVED** [review]
LGTM!

**Pull request merged**